VirtualFleetDocker.py

```diff
@@ -0,0 +1,257 @@
+    runningContainers.append(mqtt_broker_container)
+
+def start_containers(docker_client, settings, vehicle, port):
+    create_config_files(vehicle)
+    vehicle_id = f"{vehicle['company']}-{vehicle['name']}"
+
+    external_server_image_tag = f"{settings['external-server-docker-image']}:{settings['external-server-docker-tag']}"
+    external_server_container = docker_client.containers.run(
+        external_server_image_tag,
+        detach=True,
+        auto_remove=False,
+        network_mode="host",
+        volumes={
+            os.path.abspath(TMP_CONFIG_PATH): {"bind": "/home/bringauto/config", "mode": "ro"},
+            os.path.abspath("."): {"bind": "/home/bringauto/log", "mode": "rw"},
+        },
+        entrypoint=[
+            "python3",
+            "/home/bringauto/external_server/external_server_main.py",
+            f"--config=/home/bringauto/config/{vehicle_id}-external-server-config.json",
+        ],
+    )
+    logging.info(f"Started an external server docker container with id {external_server_container.short_id}")
+    runningContainers.append(external_server_container)
+
+    gateway_image_tag = f"{settings['gateway-docker-image']}:{settings['gateway-docker-tag']}"
+    gateway_container = docker_client.containers.run(
+        gateway_image_tag,
+        detach=True,
+        auto_remove=False,
+        network_mode="host",
+        volumes={
+            os.path.abspath(TMP_CONFIG_PATH): {"bind": "/home/bringauto/config", "mode": "ro"},
+            os.path.abspath("."): {"bind": "/gateway/log", "mode": "rw"},
+        },
+        entrypoint=[
+            "/home/bringauto/module-gateway/bin/module-gateway-app",
+            f"--config-path=/home/bringauto/config/{vehicle_id}-gateway-config.json",
+            f"--port={port}",
+            "--verbose",
+        ],
+    )
+    logging.info(f"Started a gateway docker container with id {gateway_container.short_id}")
+    runningContainers.append(gateway_container)
+
+    vehicle_image_tag = f"{settings['vehicle-docker-image']}:{settings['vehicle-docker-tag']}"
+    vehicle_container = docker_client.containers.run(
+        vehicle_image_tag,
+        detach=True,
+        auto_remove=False,
+        network_mode="host",
+        volumes={
+            os.path.abspath(os.path.join(CONFIG_PATH, "virtual-vehicle")): {"bind": "/virtual-vehicle-utility/config", "mode": "ro"},
+            os.path.abspath("."): {"bind": "/virtual-vehicle-utility/log", "mode": "rw"},
+        },
+        entrypoint=[
+            "/virtual-vehicle-utility/bin/virtual-vehicle-utility",
+            "--config=/virtual-vehicle-utility/config/config.json",
+            "--verbose",
+            "--module-gateway-ip=0.0.0.0",
+            f"--module-gateway-port={port}",
+        ],
+    )
+
+    logging.info(f"Started a vehicle docker container with id {vehicle_container.short_id}")
+    runningContainers.append(vehicle_container)
+
+def exit_gracefully(signum, frame):
```

Comment:
unused variables

Reply:
It needs to have those arguments because `signal.signal(signal.SIGINT, exit_gracefully)` requires them.